platform/asterix/boot/src/drivers/pmic.c

```diff
@@ -91,94 +141,83 @@ int pmic_init(void) {
+    { BUCK_BASE, BUCK1NORMVOUT, BUCKVOUT_1V8 },
+    { BUCK_BASE, BUCK1RETVOUT, BUCKVOUT_1V8 },
+    { BUCK_BASE, BUCK2NORMVOUT, BUCKVOUT_3V0 },
+    { BUCK_BASE, BUCK2RETVOUT, BUCKVOUT_3V0 },
+    { BUCK_BASE, BUCKENCTRL, 0 },
+    { BUCK_BASE, BUCKVRETCTRL, 0 },
+    { BUCK_BASE, BUCKPWMCTRL, 0 },
+    { BUCK_BASE, BUCKSWCTRLSET, BUCKSWCTRLSET_BUCK1SWCTRLSET | BUCKSWCTRLSET_BUCK2SWCTRLSET /* use registers rather than resistor settings */ },
+    { BUCK_BASE, BUCKCTRL0, 0 },
+
+    // Configure charger (TODO: values are board/battery dependent)
+    // - Thermistor: 10K NTC
+    // - Termination voltage: 4.2V
+    // - Reduced termination voltage (for warm region): 4.00V
+    // - Charge current limit of 152 mA (approximately 1C for most reasonable wearable batteries)
+    // - Discharge current limit of 200 mA (increase current measurement accuracy)
+    // - Release charger from error state if applicable (but do not clear
+    //   safety timers) -- this doesn't happen in a loop because after we
+    //   fail to boot three times, we will sit at sadwatch until a button is
+    //   pressed
+    // - Enable charging
+    { VBUSIN_BASE, VBUSINILIMSTARTUP, VBUSLIM_500MA }, // should be default, but 'reset value from OTP, value listed in this table may not be correct'
+    { CHARGER_BASE, BCHGENABLECLR, ENABLECHARGING_DISABLECHG },
+    { ADC_BASE, ADCNTCRSEL, ADCNTCRSEL_10K },
+    { CHARGER_BASE, BCHGVTERM, BCHGVTERMNORM_4V20 },
+    { CHARGER_BASE, BCHGVTERMR, BCHGVTERMREDUCED_4V00 },
+    { CHARGER_BASE, BCHGISETMSB, 38 },
+    { CHARGER_BASE, BCHGISETDISCHARGEMSB, 42 },
+    { CHARGER_BASE, TASKCLEARCHGERR, 1 },
+    { CHARGER_BASE, TASKRELEASEERROR, 1 },
+    { CHARGER_BASE, BCHGENABLESET, ENABLECHARGING_ENABLECHG },
+
+    // LDO1 as LDO @ 1.8V (powers the DA7212 ... do not back power it through I/O pins, and it must always be on because sensors share I2C bus with it!)
+    { LDSW_BASE, LDSW1GPISEL, 0 },
+    { LDSW_BASE, LDSW1VOUTSEL, LDSWVOUTSEL_1V8 },
+    { LDSW_BASE, LDSW1LDOSEL, LDSWLDOSEL_LDO },
+    { LDSW_BASE, TASKLDSW1SET, 0x01U },
+
+    // LDO2 as LDO @ 1.8V (powers the QSPI flash)
+    { LDSW_BASE, LDSW2GPISEL, 0 },
+    { LDSW_BASE, LDSW2VOUTSEL, LDSWVOUTSEL_1V8 },
+    { LDSW_BASE, LDSW2LDOSEL, LDSWLDOSEL_LDO },
+    { LDSW_BASE, TASKLDSW2SET, 0x01U },
+    
+    // GPIO0 is nPM_RST input (from debug header).  Sadly for us, that's not
+    // how nPM1300 works: you can't wire GPIO0 to be a reset, you can only
+    // wire it to be part of a two-finger reset.  So we just set it up to
+    // generate an event on falling edge, and I guess nRF can poll for it,
+    // assuming it doesn't otherwise get reset on its own.
+    { GPIOS_BASE, GPIOMODE0, GPIOMODE_GPIEVENTFALL },
+    { GPIOS_BASE, GPIOPUEN0, 1 },
+    
+    // GPIO1 is nPM interrupt output to nRF (but nRF must enable PU if it wants it, to save on Iddq)
+    { GPIOS_BASE, GPIOMODE1, GPIOMODE_GPOIRQ },
+    { GPIOS_BASE, GPIOOPENDRAIN0, 1 },
```

Comment:
probably irrelevant as bootloader does not handle any PMIC event. firmware can take care of this

Reply:
Agreed, left everything as an input.